### PR TITLE
Feature/intlar 210 custom file name

### PR DESCRIPTION
**Pull request by lukasdewijn**
- **Custom file name toggle**: Add `use_custom_file_name` boolean column and   
   toggle field so users can choose whether to serve assets with a custom  
   download file name. The text input for the custom file name is only visible
   when the toggle is enabled.
   - **Download action**: Add `DownloadAssetAction` so assets can be downloaded
   directly from the table row actions and the edit page header.
   - **Safe file name handling**: Add `Asset::buildSafeFileName()` which
   sanitizes file names by stripping dangerous characters and correctly handling
   extensions. Move download file name logic from
   `AssetController::resolveDownloadFileName()` into
   `Asset::getDownloadFileName()` so both the controller and download action
   share the same logic.
   - **Refactor CopyUrlAction**: Inline the `HasCopyUrlAction` trait directly
   into `CopyUrlAction` and remove the now-unnecessary trait and duplicate
   `CopyUrlHeaderAction` class.
   - **Asset preview fallback**: Show a document icon with extension label for
   non-previewable file types in the table preview column.
   - **Translations**: Add labels for the toggle and download action in EN, FR,
   and NL.
   - **Tests**: Add `BuildSafeFileNameTest` covering extension handling,
   dangerous character stripping, and edge cases.

   ## Test plan
   - [ ] Verify the custom file name toggle shows/hides the text input
   - [ ] Verify downloading an asset works from table and edit page
   - [ ] Verify assets without a custom file name download with the asset name
   - [ ] Verify the copy URL action still works as before
   - [ ] Verify non-previewable files show the fallback icon in the table
   - [ ] Run `php artisan test --filter=BuildSafeFileNameTest`"

### CODE REVIEW

**Review by sten — COMMENTED** [review]
todo: The first commit (04cf552) has message . — this should be squashed before merging.

File: database/migrations/add_custom_file_name_to_assets_table.php.stub [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration
+{
+    public function up()
+    {
+        Schema::table('assets', function (Blueprint $table) {
+            $table->string('custom_file_name')->nullable();
```
Comment (sten) [inline database/migrations/add_custom_file_name_to_assets_table.php.stub]: suggestion: ik zou dit vertaalbaar maken. Ik denk dat we het daarover gehad hebben.

File: src/Models/Asset.php [hunk 2]
```diff
@@ -71,6 +77,33 @@ public function getViewUrl(?string $locale = null): string
...
+            return self::buildSafeFileName($this->name, $extension);
+        }
+
+        return $this->buildSafeFileName($this->custom_file_name, $extension);
+    }
+
+    public static function buildSafeFileName(string $filename, string $extension): string
+    {
+        $filename = str_replace('..', '', $filename);
+        $filename = preg_replace('/[\/\\\\;&`<>\x00|]/', '', $filename);
```
Comment (sten) [inline src/Models/Asset.php]: todo: zet hier een comment boven van wat je allemaal uit filtert om het leesbaar te maken.

**Review by sten — CHANGES_REQUESTED** [review]
praise: nice work! 

Maar kan je door alle opmerkingen gaan? PR #18 review ik nu nog niet want dat is een mix van dit en een vorige PR. Graag eerst dit mergen en dan dit overdragen op PR #18.

En graag ook een test voor AssetController om te zien of de custom file name juist gedownload wordt.

File: src/Http/Controllers/AssetController.php [hunk 3]
```diff
@@ -54,4 +75,36 @@ private function getAssetMedia(Asset $asset, ?string $locale = null): ?Media
 
         return $assetMedia;
     }
+
+    private function getMediaPath(Media $media, ?string $conversion): string
+    {
+        if ($conversion) {
```
[the remark this answers is not in the document]
Comment (sten) [reply src/Http/Controllers/AssetController.php]: Ik denk hierbij aan een oplossing waarbij je de file_name van het media object verandert met de custom file name. En dan kan je dat media object teruggeven want dat is een Responsable.

**Review by vanneszias — COMMENTED** [review]
looks good to me!
ben nog niet helemaal vertrouwd met deze package, dus kan niet onmiddellijk in-depth analyse uitvoeren, maar op eerste opzicht lijkt dit best goed. paar kleine vraagjes nog toegevoegd.
heb dit enkel manueel nagekeken, dus nergens getest ofzo, maar ik ga ervan uit dat je dit zelf hebt gedaan?

File: database/migrations/add_custom_file_name_to_assets_table.php.stub [hunk 4]
```diff
@@ -0,0 +1,15 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration
+{
+    public function up()
```
[the remark this answers is not in the document]
Comment (lukasdewijn) [reply database/migrations/add_custom_file_name_to_assets_table.php.stub:9]: Done

File: src/Filament/Resources/AssetResource.php [hunk 5]
```diff
@@ -61,9 +64,14 @@ public static function getNavigationSort(): ?int
...
-            AssetNameField::create(true),
+            Group::make([
+                AssetNameField::create(true),
+                AssetCustomFileNameField::create(),
+            ])->columnSpan(1),
+
             AssetMediaField::create(FilamentFlexibleBlocksAssetManagerConfig::hasTranslatableAssets())
-                ->required(),
+                ->required()
+                ->columnSpan(1),
```
[the remark this answers is not in the document]
Comment (lukasdewijn) [reply src/Filament/Resources/AssetResource.php]: Weg gehaald

File: src/Models/Asset.php [hunk 6]
```diff
@@ -28,6 +30,10 @@ class Asset extends Model implements HasMedia, HasTranslatableMedia, Linkable
```
[the remark this answers is not in the document]
Comment (lukasdewijn) [reply src/Models/Asset.php:30]: done

File: src/Models/Asset.php [hunk 7]
```diff
@@ -71,6 +77,33 @@ public function getViewUrl(?string $locale = null): string
         ]);
     }
 
+    public function getDownloadFileName(): ?string
+    {
+        $media = $this->getLocalizedAssetMedia();
```
[the remark this answers is not in the document]
Comment (lukasdewijn) [reply src/Models/Asset.php:82]: Done!

File: src/Models/Asset.php (hunk 2 above)
Comment (lukasdewijn) [reply src/Models/Asset.php:101]: Done!

File: resources/lang/en/filament-flexible-blocks-asset-manager.php [hunk 8]
```diff
@@ -5,6 +5,8 @@
     'asset_lbl' => 'Asset',
     'asset_plural_lbl' => 'Assets',
     'form_component.name_lbl' => 'Name',
```
[the remark this answers is not in the document]
Comment (lukasdewijn) [reply resources/lang/en/filament-flexible-blocks-asset-manager.php]: done

File: resources/views/tables/columns/asset-preview.blade.php [hunk 9]
```diff
@@ -37,4 +38,14 @@
             />
         </div>
     </div>
-@endif
\ No newline at end of file
+    @else
+    <div style="position: relative; width: 2.5rem; height: 2.5rem; display: flex; align-items: center; justify-content: center; margin-left: 3rem;">
+        <x-heroicon-o-document style="width: 2.5rem; height: 2.5rem; color: #212121;" />
```
[the remark this answers is not in the document]
Comment (lukasdewijn) [reply resources/views/tables/columns/asset-preview.blade.php]: Hardcoded kleur verwijderd. Het icoon erft nu de standaard tekstkleur. Werkt automatisch met darkmode.

File: resources/views/tables/columns/asset-preview.blade.php [hunk 10]
```diff
@@ -37,4 +38,14 @@
...
+        @if($media?->extension)
+            <span style="position: absolute; font-size: 0.4rem; font-weight: bold; text-transform: uppercase; color: #212121; margin-top: 0.25rem;">
```
[the remark this answers is not in the document]
Comment (lukasdewijn) [reply resources/views/tables/columns/asset-preview.blade.php]: Hardcoded kleur verwijderd. Het icoon erft nu de standaard tekstkleur. Werkt automatisch met darkmode.

File: resources/views/tables/columns/asset-preview.blade.php [hunk 11]
```diff
@@ -37,4 +38,14 @@
             />
         </div>
     </div>
-@endif
\ No newline at end of file
+    @else
```
[the remark this answers is not in the document]
Comment (lukasdewijn) [reply resources/views/tables/columns/asset-preview.blade.php]: done

File: src/Filament/Form/Fields/AssetMediaField.php [hunk 12]
```diff
@@ -24,7 +25,47 @@ public static function create(bool $translatable = false): SpatieMediaLibraryFil
                 /** @var Asset $record */
                 return $record->getAssetCollection();
             })
-            ->conversion('thumbnail');
+            ->conversion('thumbnail')
+            ->getUploadedFileUsing(static function (SpatieMediaLibraryFileUpload $component, string $file): ?array {
```
[the remark this answers is not in the document]
Comment (lukasdewijn) [reply src/Filament/Form/Fields/AssetMediaField.php]: verwijderd. getUploadedFileUsing() override is weggehaald.

File: src/Models/Asset.php [hunk 13]
```diff
@@ -71,6 +77,33 @@ public function getViewUrl(?string $locale = null): string
...
+        $extension = $media?->extension;
+
+        if (! $this->use_custom_file_name || ! $this->custom_file_name) {
+            return self::buildSafeFileName($this->name, $extension);
```
[the remark this answers is not in the document]
Comment (lukasdewijn) [reply src/Models/Asset.php]: done!

File: src/Http/Controllers/AssetController.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (lukasdewijn) [reply src/Http/Controllers/AssetController.php]: getMediaPath() is verwijderd. De custom file name wordt nu op $assetMedia->file_name gezet, waarna het media object via toInlineResponse() (Responsable) wordt teruggegeven

File: database/migrations/add_custom_file_name_to_assets_table.php.stub (hunk 1 above)
Comment (lukasdewijn) [reply database/migrations/add_custom_file_name_to_assets_table.php.stub]: done

**Comment by sten** [comment]
@lukasdewijn Looks good now. Ik merge.
